README.md

```diff
@@ -1,14 +1,175 @@
 # svelte-concurrency
 
-Handle async task with ease thanks to `svelte-concurrency`
+Handle async tasks with ease thanks to `svelte-concurrency`.
+
+## What is it?
+
+`svelte-concurrency` supplies a simple way to introduce cancellable concurrency into your app. Not only do they provide the cancellability that is missing from normal Promises, `svelte-concurrency` also provides a public API that allows you to observe the running state of your task without having to set a single flag manually.
+
+Tasks that live on components are automatically cancelled when their context is destroyed, meaning you don't need to worry about the clean up - we've got you covered.
+
+Choose whether you want to keep the oldest, keep the newest or keep all instances of your task to help boost the performance of your app and reduce unnecessary server load.
+
+## How to use `svelte-concurrency`
+
+Install it using your favourite package manager:
+
+```bash
+	pnpm install svelte-concurrency
+```
+
+Then put it to work immediately wherever you want cancellable promises.
+
+## Task structure
+
+All tasks will return a store with the same structure:
+
+- error: Error - if an error occurred, it will be returned here,
+- results: Array - all of the results from previous invocations of this task,
+- last_successful: Any - the return value from the last successful run of the task
+- is_loading: Boolean - whether the task is currently running on not
+
+## Task types
+
+There are several flavours of tasks to choose from (check out the interactive docs here [insert link to site when ready]).
+
+With all types of task, it is possible to invoke it directly or add the `kind` parameter to the options object.
+
+### Standard/default task
+
+This simply gives you a task wrapper around your function. It will not handle any kind of concurrency for you.
+
+```ts
+<script>
+	import { task } from 'svelte-concurrency';
+
+	const my_task = task(async function* (param: number) {
```

Comment:
For the docs i think we should probably start with async Promises in the examples and talk about generators after having introduced the concept of mid run cancelability. If we want to use async generator this is still missing a `yield` after the await 😄